docling/datamodel/base_models.py

```diff
@@ -166,6 +166,12 @@ class DoclingComponentType(str, Enum):
     USER_INPUT = "user_input"
 
 
+class StopReason(str, Enum):
```

Comment:
let's be more explicit with the name, e.g. `GenerationStopReason` or `VlmStopReason`.